typings/single-spa.d.ts

```diff
@@ -65,18 +65,31 @@ declare module "single-spa" {
   export function setUnmountMaxTime(time: number, dieOnTimeout?: boolean): void;
   export function setUnloadMaxTime(time: number, dieOnTimeout?: boolean): void;
 
+  type Application<T = {}> =
+    | LifeCycles<T>
+    | ((config: T & AppProps) => Promise<LifeCycles<T> | Splat<LifeCycles<T>>>);
+
+  type Activity = (location: Location) => boolean;
+
+  export type RegisterApplicationConfig<T = {}> = {
+    name: string;
+    app: Application<T>;
+    activeWhen: Activity;
+    customProps?: T;
+  };
+
   // ./applications/apps.js
   export function registerApplication<T extends object = {}>(
-    appName: string,
-    applicationOrLoadingFn:
-      | LifeCycles<T>
-      | ((
-          config: T & AppProps
-        ) => Promise<LifeCycles<T> | Splat<LifeCycles<T>>>),
-    activityFn: (location: Location) => boolean,
+    appNameOrObjectConfiguration: string,
```

Comment:
Since this is an overloaded function, we don't need to worry about this first definition ever being used when the first argument is a configuration object.

```suggestion
    appName: string,
```